fili-core/src/main/java/com/yahoo/bard/webservice/web/JobsApiRequest.java

```diff
@@ -124,6 +129,46 @@ public JobsApiRequest(
     }
 
     /**
+     * Return an Observable containing a stream of job views for all the jobs in the ApiJobStore that satisfy the
+     * apiJobStoreFilters. If, for any JobRow, the mapping from JobRow to job view fails, an Observable over
+     * JobRequestFailedException is returned. If the ApiJobStore is empty, we return an empty Observable.
+     *
+     * @param apiJobStoreFilters  A Set containing apiJobStoreFilters which contain all the filter information needed
+     * to filter JobRows
+     *
+     * @return An Observable containing a stream of Maps representing the jobs to be returned to the user
+     */
+    public Observable<Map<String, String>> getFilteredJobViews(Set<ApiJobStoreFilter> apiJobStoreFilters) {
+        return apiJobStore.getFilteredRows(apiJobStoreFilters)
+                .map(this::mapJobRowsToJobViews);
+    }
+
+    /**
+     * Given a filter String, generates a Set of ApiJobStoreFilters. This method will throw a BadApiRequestException if
+     * the filter String cannot be parsed into ApiJobStoreFilters successfully.
+     *
+     * @param filterQuery  Expects a URL filterQuery String that may contain multiple filters separated by
+     * comma.  The format of a filter String is :
+     * (JobField name)-(operation)[(value or comma separated values)]?
+     *
+     * @return  A Set of ApiJobStoreFilters
+     */
+    public LinkedHashSet<ApiJobStoreFilter> buildJobStoreFilter(@NotNull String filterQuery) {
+        // split on '],' to get list of filters
+        return Arrays.asList(filterQuery.split(COMMA_AFTER_BRACKET_PATTERN)).stream()
```

Comment:
Can be replaced by `Arrays.stream(filterQuery.split(COMMA_AFTER_BRACKET_PATTERN))`